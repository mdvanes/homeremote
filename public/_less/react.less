// for /react test

// Toggle
.btn-icon {
    font-size: 350%;
}

// Button-Set
.btn-group-justified {
    > a,
    > span {
        font-size: 230%;
    }
}

.margin-top {
    margin-top: 1em;
}

.log {
    color: #fff;
    height: 300px;
}