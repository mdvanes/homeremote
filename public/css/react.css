.btn-icon {
  font-size: 350%;
}
.btn-group-justified > a,
.btn-group-justified > span {
  font-size: 230%;
}
.margin-top {
  margin-top: 1em;
}
.log {
  color: #fff;
  height: 300px;
}
