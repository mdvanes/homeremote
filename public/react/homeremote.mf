CACHE MANIFEST
index.html
../library/bootstrap/css/bootstrap.min.css
../library/bootstrap/fonts/glyphicons-halflings-regular.woff2
../library/bootstrap/fonts/glyphicons-halflings-regular.woff
../library/bootstrap/fonts/glyphicons-halflings-regular.ttf
../css/react.css
build/bundle.js

NETWORK:
*
