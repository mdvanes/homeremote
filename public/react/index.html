<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>React</title>
    
    
  </head>
  <body>
    
    <script src="build/bundle.js"></script>
    
  </body>
</html>
