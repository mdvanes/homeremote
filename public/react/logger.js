/**
 * Created by m.van.es on 6-2-2016.
 */

let logger = {};

let writeLog = (...messages) => {
    let time = new Date();
    let timestamp = time.getHours() + ':' +
        time.getMinutes() + '.' +
        time.getSeconds();
    document.querySelector('.log').value = timestamp + ' ' +
        messages.join(' ') + '\n' +
        document.querySelector('.log').value;
};

logger.log = message => {
    writeLog('INFO:    ', message);
};

logger.error = message => {
    writeLog('ERROR:', message);
};

export default logger;