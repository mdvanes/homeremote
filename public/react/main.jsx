// from https://github.com/code0wl/react-example-es2015/blob/master/app/main.jsx#L5
//import './stylesheets/main.css';
import React from 'react';
import {render} from 'react-dom';
////import FilterableProductTable from './src/filterable-product-table';
import ButtonGroup from './components/button-group';
import Toggle from './components/toggle';
import Log from './components/log';

// init shell
renderShell();

// TODO toggle as single button with icon (instead of 2 buttons)

function renderShell() {
    var shell = document.createElement('div');
    shell.className = 'app-shell container';
    document.body.appendChild(shell);
    render(
        <div>
            <div className="row">
                <div className="col-xs-12">
                    <ButtonGroup label=" 1" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 2" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 3" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 4" icon="lamp" id="clickstub"/>
                </div>
            </div>
            <div className="row margin-top">
                {/*<Toggle label=" 2" id="togglestub" icon="lamp"/>
                <Toggle label=" 3" id="togglestub" icon="lamp"/>
                <Toggle label="Switch 4" id="togglestub"/>*/}
                <Toggle label="" id="togglestub" icon="volume-up"/>{/* Broadcast */}
                <Toggle label="" id="togglestub" icon="music"/>{/* Music on server */}
            </div>
            <div className="row margin-top">
                <div className="col-xs-12">
                    <Log/>
                </div>
            </div>
        </div>, shell);
}

