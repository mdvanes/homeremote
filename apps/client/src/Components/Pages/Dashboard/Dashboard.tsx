import { Grid } from "@mui/material";
import { FC } from "react";
import DownloadList from "../../Molecules/DownloadList/DownloadList";
import LogCard from "../../Molecules/LogCard/LogCard";
import SwitchBarList from "../../Molecules/SwitchBarList/SwitchBarList";
import UrlToMusic from "../../Molecules/UrlToMusic/UrlToMusic";
import Docker from "../Docker/Docker";
import { makeStyles } from "tss-react/mui";
import DataLora from "../../Molecules/DataLora/DataLora";
import Monit from "../../Molecules/Monit/Monit";
import ServiceLinksBar from "../../Molecules/ServiceLinksBar/ServiceLinksBar";
import StreamContainer from "../../Molecules/StreamContainer/StreamContainer";
import VideoStream from "../../Molecules/VideoStream/VideoStream";
import Jukebox from "../../Molecules/Jukebox/Jukebox";
import Schedule from "../../Molecules/Schedule/Schedule";

const useStyles = makeStyles()((theme) => ({
    container: {
        "& .card-dashboard-height": {
            minHeight: "374px",
        },
        "& > .MuiGrid-item > .MuiPaper-root, & > .MuiGrid-item > .MuiContainer-root":
            {
                marginBottom: theme.spacing(2),
            },
    },
}));

const Dashboard: FC = () => {
    const { classes } = useStyles();
    return (
        <Grid container spacing={2} className={classes.container}>
            <Grid item xs={12} md={3}>
                <SwitchBarList />
                <UrlToMusic />
                <LogCard />
            </Grid>
            <Grid item xs={12} md>
                <StreamContainer />
                <Jukebox />
                <VideoStream />
                <DataLora />
            </Grid>
            <Grid item xs={12} md={5}>
                <ServiceLinksBar />
                <Docker />
                <Schedule />
                <DownloadList />
                <Monit />
            </Grid>
            <Grid item xs={12} md={2}></Grid>
        </Grid>
    );
};

export default Dashboard;
