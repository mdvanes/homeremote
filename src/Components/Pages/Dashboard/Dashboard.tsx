import { FC } from "react";
import { Box, Grid } from "@material-ui/core";
import SwitchBarList from "../../Molecules/SwitchBarList/SwitchBarList";
import LogCard from "../../Molecules/LogCard/LogCard";
import Streams from "../Streams/Streams";
import UrlToMusic from "../../Molecules/UrlToMusic/UrlToMusic";

import { makeStyles } from "@material-ui/core";
import DownloadList from "../../Molecules/DownloadList/DownloadList";

const useStyles = makeStyles(() => ({
    container: {
        "& .card-dashboard-height": {
            minHeight: "374px",
        },
    },
}));

const Dashboard: FC = () => {
    const classes = useStyles();
    return (
        <Grid container spacing={2} className={classes.container}>
            <Grid item xs={12} md={3}>
                <SwitchBarList />
            </Grid>
            <Grid item xs={12} md>
                <Streams />
            </Grid>
            <Grid item xs={12} md={3}>
                <DownloadList />
            </Grid>
            <Grid item xs={12} md={2}>
                <UrlToMusic />
                <Box marginTop={2} />
                <LogCard />
            </Grid>
        </Grid>
    );
};

export default Dashboard;
