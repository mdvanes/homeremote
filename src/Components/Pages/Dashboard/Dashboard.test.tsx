import React from "react";
import { render } from "@testing-library/react";
import Dashboard from "./Dashboard";

jest.mock(
    "../../Molecules/SwitchBarList/SwitchBarList",
    () => "mock-switch-bar-list"
);

jest.mock("../../Molecules/LogCard/LogCard", () => "mock-log-card");

jest.mock("../../Molecules/UrlToMusic/UrlToMusic", () => "mock-url-to-music");

jest.mock("@mdworld/homeremote-stream-player", () => "mock-stream-player");

describe("Dashboard page", () => {
    it("contains all the control components", () => {
        const { baseElement } = render(<Dashboard />);
        expect(baseElement).toMatchSnapshot();
    });
});
