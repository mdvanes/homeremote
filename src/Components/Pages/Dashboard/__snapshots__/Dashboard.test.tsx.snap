// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard page contains all the control components 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
      >
        <mock-switch-bar-list />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
      >
        <div
          style="font-family: Roboto; max-width: 800px;"
        >
          <mock-stream-player
            url=""
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
      >
        <mock-log-card />
      </div>
    </div>
  </div>
</body>
`;
